### DISCUSSION TITLE
Keycloak-X jgroup metrics

### DISCUSSION THREAD

**Discussion by petermazan** [keycloak/keycloak#10440] [Archive]
I successfully migrate Keycloak from version 16  to new quarkus based verison 17. I built it with KC_METRICS_ENABLED=true option and i can see that some metrics are available at /auth/metrics endpoint. However, some metrics (e.g. jgroups, infinispan) are missing there. How to build/configure Keycloak so also cluster related metrics are available? Thanks

**Comment by raul-verdi** [comment]
Similar issue here. I added more metrics to keycloak with this package https://github.com/aerogear/keycloak-metrics-spi. However, I still could not find a way to get metrics of the cluster under quarkus. 
Created an issue about it https://github.com/keycloak/keycloak/issues/11086

**Comment by christianwoehrle** [comment]
You could enable enable the  infinispan statistic in a config file like the one below (statistics=true is the relevant part) and use it with the parameter `--cache-config-file=cache-ispn-dnsping.xml` like this. 



`
 /opt/keycloak/bin/kc.sh build  --metrics-enabled=true  \
        --cache=ispn \
        --cache-config-file=cache-ispn-dnsping.xml \
        --cache-stack=kubernetes \
`

However I could not get the JGROUPS metrics with that. Just the infinispan metrics.

File cache-ispn-dnsping.xml


`
<?xml version="1.0" encoding="UTF-8"?>
<infinispan
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="urn:infinispan:config:13.0 http://www.infinispan.org/schemas/infinispan-config-13.0.xsd"
        xmlns="urn:infinispan:config:13.0">

    

    
    
    
    <cache-container name="keycloak" statistics="true">
        
        
        <transport lock-timeout="60000" />
        <local-cache name="realms" statistics="true">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <local-cache name="users" statistics="true">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <distributed-cache name="sessions" owners="2" statistics="true">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="authenticationSessions" owners="2" statistics="true">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="offlineSessions" owners="2" statistics="true">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="clientSessions" owners="2" statistics="true">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="offlineClientSessions" owners="2" statistics="true">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="loginFailures" owners="2" statistics="true">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <local-cache name="authorization" statistics="true">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <replicated-cache name="work" statistics="true">
            <expiration lifespan="-1"/>
        </replicated-cache>
        <local-cache name="keys" statistics="true">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <expiration max-idle="3600000"/>
            <memory max-count="1000"/>
        </local-cache>
        <distributed-cache name="actionTokens" owners="2" statistics="true" >
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <expiration max-idle="-1" lifespan="-1" interval="300000"/>
            <memory max-count="-1"/>
        </distributed-cache>
    </cache-container>
</infinispan>
`
